File: examples-sdk/swift/030-basic-trilane-messaging/main.swift
// 030-basic-trilane-messaging — Swift SDK equivalent
// Compile: vil compile --from swift --input 030-basic-trilane-messaging/main.swift --release

let server = VilServer(name: "ecommerce-order-pipeline", port: 8080)
let gateway = ServiceProcess(name: "gateway")
server.service(gateway)
let fulfillment = ServiceProcess(name: "fulfillment")
fulfillment.endpoint(method: "GET", path: "/status", handler: "fulfillment_status")
server.service(fulfillment)
server.compile()

File: examples-sdk/kotlin/030-basic-trilane-messaging/main.kt
// 030-basic-trilane-messaging — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 030-basic-trilane-messaging/main.kt --release

fun main() {
    val server = VilServer("ecommerce-order-pipeline", 8080)
    val gateway = ServiceProcess("gateway")
    server.service(gateway)
    val fulfillment = ServiceProcess("fulfillment")
    fulfillment.endpoint("GET", "/status", "fulfillment_status")
    server.service(fulfillment)
    server.compile()
}
